src/models/ComponentStyle.js

```diff
@@ -99,6 +102,14 @@ export default (
       return name
     }
 
+    findAndInjectKeyframes(styleSheet: StyleSheet) {
+      this.rules.forEach(rule => {
+        if (rule instanceof Keyframes) {
```

Comment:
This is highly inefficient unfortunately; we need to handle this either on a StyleSheet level (Not preferable) or extract it once we can via your class combined with componentstyle, which I think would be more elegant :)